### PR TITLE
option to not reset eval metric with Speedometer

### PR DESCRIPTION
**Pull request by geoalgo**
Speedometer reset metrics which invalidates callbacks coming after them. 

For instance

mod.fit(
    train_data=train_iter,
    eval_data=test_iter,
    eval_metric=[mx.metric.CrossEntropy(), 'acc'],
    batch_end_callback=[mx.callback.Speedometer(batch_size, 1), mx.callback.log_train_metric(1)]
)

gives the following log

INFO:root:Epoch[0] Batch [1]	Speed: 311.33 samples/sec	Train-cross-entropy=5.322994
INFO:root:Epoch[0] Batch [1]	Speed: 311.33 samples/sec	Train-accuracy=0.015625
INFO:root:Iter[0] Batch[1] Train-cross-entropy=nan
INFO:root:Iter[0] Batch[1] Train-accuracy=nan

I have set the default reset option of Speedometer to False to have the same default behavior as the other callback log_train_metric (also because it is unexpected for the user to have a default that has a side effect on the metric).

### CODE REVIEW

**Comment by geoalgo** [comment]
Could someone have a look to this pull request? It is addressing issue https://github.com/dmlc/mxnet/issues/5767

**Comment by piiswrong** [comment]
The problem with this is printed training acc will lag behind current acc because it's averaging over the low accs in the beginning of the epoch. You also will see jumps in acc in the beginning of every epoch, which can be confusing

**Comment by geoalgo** [comment]
True but dont you think printing NAN is more confusing? Maybe we could let this option but set it to false by default?

**Comment by piiswrong** [comment]
we can try to figure out a better way later. For now could you change default to auto_reset=True so that default behavior is not changed.

**Comment by piiswrong** [comment]
one way is to remove the reset from speedometer and put it in module.fit

**Comment by geoalgo** [comment]
I have set auto_reset=True as you suggested.
